typescript/src/lib/bitcoin/csuint.ts

```diff
@@ -0,0 +1,48 @@
+import { Hex } from "../utils"
+
+/**
+ * Reads the leading compact size uint from the provided variable length data.
+ *
+ * WARNING: CURRENTLY, THIS FUNCTION SUPPORTS ONLY 1-BYTE COMPACT SIZE UINTS
+ * AND WILL THROW ON COMPACT SIZE UINTS OF DIFFERENT BYTE LENGTH.
+ *
+ * @param varLenData Variable length data preceded by a compact size uint.
+ * @returns An object holding the value of the compact size uint along with the
+ *          compact size uint byte length.
+ */
```

Comment:
Do we have unit tests for this utility?

Reply:
Of course, see `bitcoin.test.ts`: https://github.com/keep-network/tbtc-v2/blob/4f766f31d784d840c9f996bc6d4cb15bd756f214/typescript/test/lib/bitcoin.test.ts#L764C45-L764C45